frontend/packages/dev-console/src/components/topology/TopologyPage.tsx

```diff
@@ -125,11 +113,54 @@ export const TopologyPageContext: React.FC<TopologyPageProps> = observer(({ matc
   );
 });
 
-export const TopologyPage: React.FC<TopologyPageProps> = ({ match }) => {
+type TopologyWorkloadsPageProps = {
+  match: RMatch<{
+    name?: string;
+  }>;
+};
+
+export const TopologyWorkloadsPage: React.FC<TopologyWorkloadsPageProps> = ({ match }) => {
+  const namespace = match.params.name;
+  const queryParams = useQueryParams();
+  const view = queryParams.get('view');
+  if (!view) {
+    setQueryArgument('view', getTopologyWorkloadsActiveView() || 'list');
+  }
+  const showGraphView = queryParams.get('view') === 'graph';
+  const onViewChange = (graphView: boolean) => {
+    if (graphView) {
+      setQueryArgument('view', 'graph');
+    } else {
+      setQueryArgument('view', 'list');
+    }
+  };
+
+  React.useEffect(() => setTopologyWorkloadsActiveView(showGraphView ? 'graph' : 'list'), [
+    showGraphView,
+  ]);
+
+  return (
+    <DataModelProvider namespace={namespace}>
+      <NamespacedPage
+        variant={showGraphView ? NamespacedPageVariants.default : NamespacedPageVariants.light}
+        hideProjects
+        toolbar={<TopologyPageToolbar showGraphView={showGraphView} onViewChange={onViewChange} />}
+      >
+        <TopologyDataRenderer showGraphView={showGraphView} />
+      </NamespacedPage>
+    </DataModelProvider>
+  );
+};
+
+export const TopologyPage: React.FC<TopologyPageProps> = ({ match, workloadsPage = false }) => {
   const namespace = match.params.name;
   return (
     <DataModelProvider namespace={namespace}>
-      <TopologyPageContext match={match} />
+      {workloadsPage ? (
+        <TopologyWorkloadsPage match={match} />
+      ) : (
+        <TopologyPageContext match={match} />
+      )}
```

Comment:
Both views are exactly the same minus the project selector.
I think they should share a lot more in common between the two pages and come up with a common strategy for the URL handling.

Reply:
Updated.